frontend/src/Utils/ModelFeatureFlagLoader.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useFeatureFlag } from '@openshift/dynamic-plugin-sdk';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+
+export type ModelFeatureFlagLoaderProps = {
+  modelFeature: {
+    flag: string;
+    model: {
+      group: string;
+      version: string;
+      kind: string;
+    };
+  };
+  namespace: string;
+};
+
+/**
+ * Component to wrap setting and loading of model feature flag.
+ * @param ModelFeatureFlagLoaderProps which model and feature flag to set for given namespace.
+ * @returns null
+ */
+const ModelFeatureFlagLoader: React.FC<ModelFeatureFlagLoaderProps> = ({ modelFeature, namespace }) => {
+  const [, setFlag] = useFeatureFlag(modelFeature.flag);
+  const [data, loaded] = useK8sWatchResource({
+    isList: false,
+    groupVersionKind: modelFeature.model,
+    namespace,
+  });
+
+  React.useEffect(() => {
+    setFlag(false);
+  }, []);
```

Comment:
I think it's important that flags get initialized to `false` before early on before any rendering happens otherwise flagged extensions will be enabled until a value of `false` is set.

Reply:
For better or worse, we differentiate between pending and false for flags in OpenShift console. Do we have a pending flag state in HAC core?

Reply:
HAd to slightly tweak this, but this assign of `false` value happens on initial render I guess that's soon enough.